src/mo_types/type.ml

```diff
@@ -1375,7 +1416,12 @@ and combine_fields rel lubs glbs fs1 fs2 =
     | _ ->
       match combine rel lubs glbs f1.typ f2.typ with
       | typ ->
-       {lab = f1.lab; typ; src = empty_src} :: combine_fields rel lubs glbs fs1' fs2'
+        let src =
```

Comment:
This seems safer to me since it's actually functional, we are constructing a new type without updating f1.src or f2.src.

Can we do the analysis without the destructive subtyping?